WordPress/Classes/ViewRelated/Jetpack/Branding/Banner/JetpackBannerView.swift

```diff
@@ -32,16 +32,26 @@ class JetpackBannerView: UIView {
 // MARK: Responding to scroll events
 extension JetpackBannerView: Subscriber {
 
-    typealias Input = CGFloat
+    typealias Input = JPScrollViewDataDelegate
     typealias Failure = Never
 
     func receive(subscription: Subscription) {
         subscription.request(.unlimited)
     }
 
-    func receive(_ input: CGFloat) -> Subscribers.Demand {
+    func receive(_ input: JPScrollViewDataDelegate) -> Subscribers.Demand {
+        let isHidden: Bool
 
-        let isHidden: Bool = input < 0
+        /// The scrollable content isn't any larger than its frame, so don't hide the banner if the view is bounced.
+        if input.scrollViewContentHeight <= input.scrollViewFrameHeight {
+            isHidden = false
+        /// Don't hide the banner until the view has scrolled down some. Currently the height of the banner itself.
+        } else if input.scrollViewContentOffsetY <= frame.height {
+            isHidden = false
+        /// The banner hides on scrolling down, shows on scrolling up.
+        } else {
+            isHidden = input.translationY < 0
+        }
```

Comment:
This seems a bit too complex for a view (`JetpackBannerView`) that should not be aware of this logic: I'd leave this logic to the sender (that is: the `scrollViewDidScroll` method) and keep `JetpackBannerView` agnostic. This is because, if we encounter another exception like reader search, we might need another ad-hoc logic, and it might very well conflict with this one. This way, we can also keep it simple and continue passing a single `CGFloat` instead of a reference to a `UIScrollView` to the subscriber. What do you think?

Reply:
Thank for your input @Gio2018! I'm not sure I follow completely, so I'll add my observation:

We already have logic in `JetpackBannerView` for the `y` translation of the scroll view, and IMO that's a good place for it. However, there are other cases such as a bouncing scroll view that we can account for with more data.

> This is because, if we encounter another exception like reader search

If I can understand this example better maybe that'll help.

> This way, we can also keep it simple and continue passing a single CGFloat instead of a reference to a UIScrollView to the subscriber

I understand this some, especially the entire reference part.

> I'd leave this logic to the sender (that is: the scrollViewDidScroll method

My concern was that we'd be repeating a lot of the show / hide logic in multiple VCs. To me it seems like a good fit for `JetpackBannerView` to decide when it should show and hide when it gets scroll view data.

Reply:
> We already have logic in JetpackBannerView for the y translation of the scroll view

exactly, in fact it probably needs to be simplified by just passing a boolean

> i f I can understand this example better maybe that'll help.

sure: the concern here is that you are adding a very specific behavior to a view that potentially can just hide or show itself (evaluating content size and offset of a referenced `UIScrollView`. What if, for any reason, tomorrow we need another specific behavior? 

> My concern was that we'd be repeating a lot of the show / hide logic in multiple VCs.

How many of these do we currently have in the app?